src/events.jl

```diff
@@ -0,0 +1,136 @@
+@with_kw mutable struct EventList{T1<:AbstractVector,T2<:AbstractVector,T3<:AbstractMatrix}
+    time::T1
+    energy::T2=Float64[]
+    ncounts::Int=0
+    mjdref::Float64=0
+    dt::Float64=0
+    notes::String=""
+    gti::T3=reshape(Float64[],0,2)
+    PI::Vector{Int64}=Int[]
+    high_precision::Bool=false
+    mission::String=""
+    instr::String=""
+    header::String=""
+    detector_id::Vector{String}=String[]
+    ephem::String=""
+    timeref::String=""
+    timesys::String=""
+end
+
+function read(filename::String, format::String)
+    if format=="fits"
+        return load_events_from_fits(filename)
+    else
+        throw(ArgumentError("File not supported still."))
```

Comment:
I'd make the error message more explicit:
```suggestion
        throw(ArgumentError("File format $(format) not yet supported."))
```
Same below.